### PR TITLE
Enhanced Test Plan Detection and Reporting

### PR DESCRIPTION
**Pull request by rusel95**
### Summary

This PR enhances the ReportPortal XCTest Agent to automatically detect and include test plan names in launch names, similar to how Android's ReportPortal integration displays suite names. Additionally, it improves stack trace formatting and fixes various issues related to metadata collection.

### Changes Made

  1. Test Plan Name in Launch Names
  - Added automatic detection of test plan names via TEST_PLAN_NAME environment variable
  - Test plan names are now included in ReportPortal launch names (e.g., LaunchName: TestPlanName)
  - Spaces in test plan names are automatically replaced with underscores for better
  compatibility
  - Simple, reliable implementation using only environment variables (no file system operations)

  2. Enhanced Stack Trace Formatting

  - Improved error message formatting with clear structure and headers
  - Added safety bounds to the enhanceStackFrame method to prevent crashes on malformed input
  - Limited string lengths and array operations for better performance and security
  - Enhanced stack frame parsing with proper validation

  3. Tag Parsing Fix

  - Fixed issue where tags with spaces weren't properly parsed
  - Each tag is now trimmed individually after splitting by a comma
  - Tags now display correctly as individual badges in ReportPortal

  4. Metadata Collection Improvements

  - Simplified test plan detection to use only environment variables
  - Removed complex file system operations for better reliability in CI/CD
  environments
  - Maintained all existing device metadata collection

  5. iOS Deployment Target Update

  - Updated minimum iOS deployment target from 11.4 to 12.0 to resolve Xcode warnings

  Configuration

  Users can configure test plan names by adding to their .xctestplan file:

```
  {
    "defaultOptions": {
      "environmentVariableEntries": [
        {
          "key": "TEST_PLAN_NAME",
          "value": "Smoke Tests"
        }
      ]
    }
  }

```
Result in ReportPortal: YourLaunchName: Smoke_Tests

CI/CD environments can override: TEST_PLAN_NAME="Nightly Tests" xcodebuild test ...

Testing

  - Builds successfully with no warnings
  - Test plan names appear correctly in ReportPortal launches
  - Handles edge cases (spaces, empty values, missing environment variables)
  - Works reliably across local development and CI environments

_Migration Guide_

  No migration needed. To use the new test plan feature, simply add the TEST_PLAN_NAME environment variable to your test plan configuration.

### CODE REVIEW

File: Sources/Utilities/MetadataCollector.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        
+        // OS version - handle iPadOS separately
+        var osName = UIDevice.current.systemName
+        if osName == "iOS" && UIDevice.current.userInterfaceIdiom == .pad {
+            // iPad running iOS 13+ reports as "iOS" but is actually iPadOS
+            let osVersionComponents = UIDevice.current.systemVersion.split(separator: ".")
+            if let majorVersion = osVersionComponents.first, let major = Int(majorVersion), major >= 13 {
+                osName = "iPadOS"
+            }
+        }
```
[the remark this answers is not in the document]
Comment (HardNorth) [reply Sources/Utilities/MetadataCollector.swift]: Good point, please fix.

File: Sources/ReportingService.swift [hunk 2]
```diff
@@ -358,57 +424,310 @@ private extension ReportingService {
...
+        }
+        
+        // Remove common prefixes
+        if demangled.hasPrefix("_T") {
+            demangled = String(demangled.dropFirst(2))
+        }
+        
+        // Final length limit
+        return String(demangled.prefix(200))
     }
```
[the remark this answers is not in the document]
Comment (HardNorth) [reply Sources/ReportingService.swift]: Not sure if this is feasible, so please consider on your own.

**Review by HardNorth — CHANGES_REQUESTED** [review]
Please resolve Copilot review comments.

File: Sources/ReportingService.swift [hunk 3]
```diff
@@ -30,22 +31,28 @@ public class ReportingService {
...
     httpClient.setPlugins([AuthorizationPlugin(token: configuration.portalToken)])
   }
   
   private func getStoredLaunchID(completion: @escaping (String?) -> Void) throws {
     let endPoint = GetCurrentLaunchEndPoint()
-    try httpClient.callEndPoint(endPoint) { (result: LaunchListInfo) in
-      guard let launch = result.content.first, launch.status == "IN_PROGRESS" else {
-        completion(nil)
-        return
+    do {
```
[the remark this answers is not in the document]
Comment (rusel95) [reply Sources/ReportingService.swift]: actually, this makes sense. fixed.

File: Sources/Utilities/MetadataCollector.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (rusel95) [reply Sources/Utilities/MetadataCollector.swift]: fixed. thanks.

File: Sources/ReportingService.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (rusel95) [reply Sources/ReportingService.swift]: It's not 100% required but rather a nice-to-have, but it doesn't bring noticeable value, so I've removed it all.

**Review by HardNorth — APPROVED** [review]
LGTM
